sw/device/tock/meson.build

```diff
@@ -18,6 +20,33 @@ rust_flags = '-C link-arg=-Tlayout.ld ' + \
 
 build_tock_cmd = meson.source_root() / 'util/build_tock.sh'
 
+llvm_tools_installed = false
+rust_src_installed = false
+target_tools_installed = false
+
+# Query rustup and search for required components
+rustup_components = run_command(rustup, 'component', 'list', '--toolchain', toolchain).stdout()
+foreach c : rustup_components.split('\n')
```

Comment:
What about:

```python
foreach c : rustup_components.split('\n').filter(|x| x.endswith('(installed)')
```

And then you can drop the repeated `endswith` check?

Reply:
I don't believe meson has a filter method for arrays https://mesonbuild.com/Syntax.html#array-methods

Reply:
Yep. Meson does not support doing anything fancy with arrays.